website/docs/metaprogramming-plugins.md

```diff
@@ -0,0 +1,124 @@
+---
+id: metaprogramming-plugins
+title: Metaprogramming plugins
+---
+
+> **Note**: The plugin system has many drawbacks and is not considered a stable
+> feature. Expect its API to change in the future.
```

Comment:
Do you mind bringing this language in line with what the other docs use? I would use the words "known limitations" and "experimental" somewhere.

**T.self_type**

> **Warning**: This feature is experimental and has known limitations. It may
> not work as expected or change without notice.

**Tuples**

> **Warning**: Tuples have many known limitations, and should be considered an
> experimental feature. They may not work as expected or change without notice.

**Shapes**

> **Warning**: Shape types have many known limitations, and should be considered
> an experimental feature. They may not work as expected or change without
> notice. For an alternative that plays better with static type checking, see
> [Typed Structs](tstruct.md).